src/Microsoft.Extensions.Configuration.AzureAppConfiguration/JsonConfiguration/JsonConfigurationParser.cs

```diff
@@ -0,0 +1,115 @@
+                if (type == FeatureManagementConstants.ContentType ||
+                    type == KeyVaultConstants.ContentType)
+                {
+                    return false;
+                }
+
+                if (type.Contains('/'))
+                {
+                    string mainType = type.Split('/')[0];
+                    string subType = type.Split('/')[1];
+                    if (mainType == acceptedMainType)
+                    {
+                        if (subType.Contains('+'))
+                        {
+                            var subTypes = subType.Split('+');
+                            if (Array.Exists(subTypes, x => x == acceptedSubType))
+                            {
+                                return true;
+                            }
+                        }
+                        else if (subType == acceptedSubType)
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            catch (Exception ex) when (ex is ArgumentException || ex is FormatException)
+            {
+                // not a valid JSON content type
+            }
+            return false;
+        }
+
+        public static List<KeyValuePair<string, string>> ParseJsonSetting(ConfigurationSetting setting)
+        {
+            List<KeyValuePair<string, string>> keyValues = new List<KeyValuePair<string, string>>();
+            Dictionary<string, string> keyValueDict = new Dictionary<string, string>();
+            ParseSetting(setting.Key, setting.Value, keyValueDict);
+            foreach (KeyValuePair<string, string> entry in keyValueDict)
+            {
+                keyValues.Add(entry);
+            }
+            return keyValues;
+        }
+
+
+        public static void ParseSetting(string currentKey, string currentValue, Dictionary<string, string> keyValueDict)
+        {
+            try
+            {
+                var json_data = JsonSerializer.Deserialize<JsonElement>(currentValue);
+                switch (json_data.ValueKind)
+                {
+                    case JsonValueKind.Array:
+                        var valueArray = json_data.EnumerateArray();
+                        for (int index = 0; index < valueArray.Count(); index++)
+                        {
+                            var newKey = ConfigurationPath.Combine(new List<string> { currentKey, index.ToString() });
+                            ParseSetting(newKey, valueArray.ElementAt(index).GetRawText(), keyValueDict);
+                        }
+                        break;
+
+                    case JsonValueKind.Object:
+                        var valueObject = json_data.EnumerateObject();
+                        foreach(JsonProperty entry in valueObject)
+                        {
+                            var newKey = ConfigurationPath.Combine(new List<string> { currentKey, entry.Name.ToString() });
```

Comment:
>entry.Name [](start = 98, length = 10)

Could `entry.Name` be `null`?

Reply:
Removed the ToString() call because entry.Name  is already string.

---
In reply to: [454622297](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r454622297) [](ancestors = 454622297)